### ISSUE TITLE
Error using LBaaS with OpenStack

### ISSUE THREAD

**Issue by mewald1** [hashicorp/terraform#5380]
Hi, I am using the following code (excerpt) to use LBaaS:

```
resource "openstack_lb_monitor_v1" "mon_webserver" {
  type = "TCP"
  delay = 30
  timeout = 5
  max_retries = 3
  admin_state_up = "true"
}

resource "openstack_lb_pool_v1" "pool_webservers" {
  name = "webservers"
  protocol = "TCP"
  subnet_id = "${openstack_networking_subnet_v2.snet-fe.id}"
  lb_method = "ROUND_ROBIN"
  monitor_ids = ["${openstack_lb_monitor_v1.mon_webserver.id}"]
}

resource "openstack_lb_member_v1" "member_web01" {
  pool_id = "${openstack_lb_pool_v1.pool_webservers.id}"
  address = "${openstack_compute_instance_v2.web01.access_ip_v4}"
  port = 80
}

resource "openstack_lb_member_v1" "member_2" {
  pool_id = "${openstack_lb_pool_v1.pool_webservers.id}"
  address = "${openstack_compute_instance_v2.web02.access_ip_v4}"
  port = 80
}

resource "openstack_lb_vip_v1" "vip_web" {
  name = "web"
  subnet_id = "${openstack_networking_subnet_v2.snet-fe.id}"
  protocol = "TCP"
  port = 80
  pool_id = "${openstack_lb_pool_v1.pool_webservers.id}"
}
```

I am getting the following error:

```
Error applying plan:

2 error(s) occurred:

* openstack_lb_member_v1.member_2: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.12
    Resource ID: openstack_lb_member_v1.member_2
    Mismatch reason: attribute mismatch: address
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"address":*terraform.ResourceAttrDiff{Old:"", New:"${openstack_compute_instance_v2.web02.access_ip_v4}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "port":*terraform.ResourceAttrDiff{Old:"", New:"80", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "weight":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "admin_state_up":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "pool_id":*terraform.ResourceAttrDiff{Old:"", New:"${openstack_lb_pool_v1.pool_webservers.id}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"port":*terraform.ResourceAttrDiff{Old:"", New:"80", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "weight":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "admin_state_up":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "pool_id":*terraform.ResourceAttrDiff{Old:"", New:"8fbcecd0-7e10-4e05-82d8-0619b6e1c4a5", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}}, Destroy:false, DestroyTainted:false}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.

* openstack_lb_member_v1.member_web01: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.12
    Resource ID: openstack_lb_member_v1.member_web01
    Mismatch reason: attribute mismatch: address
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"pool_id":*terraform.ResourceAttrDiff{Old:"", New:"${openstack_lb_pool_v1.pool_webservers.id}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "address":*terraform.ResourceAttrDiff{Old:"", New:"${openstack_compute_instance_v2.web01.access_ip_v4}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "port":*terraform.ResourceAttrDiff{Old:"", New:"80", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "weight":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "admin_state_up":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"pool_id":*terraform.ResourceAttrDiff{Old:"", New:"8fbcecd0-7e10-4e05-82d8-0619b6e1c4a5", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "port":*terraform.ResourceAttrDiff{Old:"", New:"80", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "weight":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "admin_state_up":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.


Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

Can somebody help?

**Comment by jtopjian** [comment]
@mewald1 Thanks for the report. This bug might have been fixed yesterday. Are you able to compile from the latest source and try? See the [README](https://github.com/hashicorp/terraform/blob/master/README.md#developing-terraform) for some quick instructions on how to build from source.

If you do test and the problem is solved, can you let me know a little about your OpenStack environment? Is it an in-house built OpenStack cloud, a vendor supported one, a public cloud?

Finally, if the latest source does _not_ fix this problem, can you run Terraform in debug mode and paste the output to a gist?

```shell
$ TF_LOG=DEBUG terraform apply
```

Thanks! :)

**Comment by jtopjian** [comment]
@mewald1 just wanted to touch base on this. A new version of Terraform was released yesterday. Are you still seeing this problem with the new version?

**Comment by mitchellh** [comment]
Hello! This should be fixed in later versions of Terraform (0.7.11 at time of writing). 

Its hard to know 100% for a couple reasons:

  * This is on a now much-older version of Terraform. This is our fault, not yours, for not trying sooner, but we've fixed many diff mismatch issues since then.

  * We've fixed issues with exact error messages as this in those versions. 

Please try again! If the issue persists please just open a new issue. Thanks!
